uarray/_typing.pyi

```diff
@@ -0,0 +1,54 @@
+"""Helper module with various typing-related utilities."""
```

Comment:
The content of this small helper module could in principle be merged with `_uarray.pyi`, but having the proper type annotations separate from helper types minimizes clutter somewhat, in my opinion.